pkg/apb/provision.go

```diff
@@ -14,37 +15,47 @@ import (
 func Provision(
 	spec *Spec, parameters *Parameters,
 	clusterConfig ClusterConfig, log *logging.Logger,
-) error {
+) (*ExtractedCredentials, error) {
```

Comment:
Formatting wise? I've been favoring this when signatures start to blow up, kinda like it :)